mesonpy/__init__.py

```diff
@@ -672,7 +672,19 @@ def platform_tag(self) -> str:
             return 'any'
         # XXX: Choose the sysconfig platform here and let something like auditwheel
         #      fix it later if there are system dependencies (eg. replace it with a manylinux tag)
-        return sysconfig.get_platform().replace('-', '_').replace('.', '_')
+        platform_ = sysconfig.get_platform()
+        parts = platform_.split('-')
+        if parts[0] == 'macosx' and parts[1] in ('11', '12'):
+            # Workaround for bug where pypa/packaging does not consider Mac tags
+            # without minor versions valid. Mac started to do this in version
+            # 11, and pypa/packaging should handle things correctly from version
```

Comment:
I'd suggest adjusting this comment to say "Some Python flavors (Homebrew for example) on macOS started to do this in version 11".

Reply:
Added :blush: